astropy/io/fits/convenience.py

```diff
@@ -494,6 +494,9 @@ def table_to_hdu(table):
         fits_rec = FITS_rec.from_columns(np.array(table.filled()))
         table_hdu = BinTableHDU(fits_rec)
 
+    # FIXME: this should go somewhere in astropy.io.fits.hdu, but where?
```

Comment:
Actually, as far as Table-FITS interface goes, this might actually be the right place to put your fix. The sole purpose `table_to_hdu()` is for the unified IO function `write_table_fits()` in `connect.py`.